docs/design/README.md

```diff
@@ -0,0 +1,552 @@
+
+An additional problem to solve is that the paths to mounted directories need to be absolutely the same on the host as they are in the Twoliter container.
+This will not be problem unless the host build or sourcecode directory is in some wildly unusual place, like `/etc/docker` or `/usr/bin`.
+We can prevent this by making a certain set of top-level directories off-limits on the host for the directories we need to mount.
+An error message will describe the problem instead of running the container and failing due to occluded directories.
+
+## Twoliter Projects
+
+Twoliter will have the concept of a project.
+A project can contain multiple variant definitions, multiple kit definitions, and multiple packages.
+
+A project will have a top level `Twoliter.toml` file (and possibly a `Twoliter.lock` file).
+When `twoliter` commands are invoked, the tool will search for the `Twoliter.toml` file in `.`.
+If it doesn't find it, it will search in `..`, then `../..`, etc.
+This will allow the tool to work from any subdirectory within the project (and is inspired by `cargo`).
+
+`Twoliter.toml` will contain global project settings. Here is a sample `Twoliter.toml` file.
+
+```toml
+# This let's twoliter know if it is reading an older version of the Twoliter.toml
+# schema, or if the directory structure of the project is from an older version.
+twoliter-project-version = 1
+
+# This is the name of this project, which can have multiple variants in it. For
+# in-tree builds the project name is "bottlerocket", and thus we produce images
+# with names like `bottlerocket-aws-ecs-1` where the "project" name is a prefix
+# to the variant name. Here we would produce images with names like
+# `ootb-variants-example-dev` where `ootb-variants` is the project name and
+# `example-dev` is the variant name.
+project-name = "ootb-variants"
+
+# This allows the maintainer to rev a version globally for all variants defined in
+# the project. We may choose to enforce an opinionated versioning scheme here or
+# allow any arbitrary string, TBD.
+project-version = "1.0.0"
+
+# The container image with the SDK, buildsys, etc. This is provided as a structure
+# so that we can switch on the target architecture. For example, when building
+# Bottlerocket images for x86_64 machines, the following would be produced:
+# public.ecr.aws/bottlerocket/bottlerocket-build-tools-x86_64:v1.14.1. Twoliter
+# will assume the container is available locally if the registry field is null
+# or an empty string.
+[sdk-image]
+registry = "public.ecr.aws/bottlerocket"
+name = "bottlerocket-sdk"
+version = "v0.30.0"
+
+# This is the container that Twoliter uses to invoke `cargo make` and other build
+# commands. It will be available for both x86_64 and Arm hosts, but it is agnostic
+# as to which architecture the build is targeting.
+[twoliter-image]
+registry = "public.ecr.aws/bottlerocket"
+name = "twoliter"
+version = "v0.1.0"
+```
```

Comment:
Does this have a dependency on the `sdk-image`?

```suggestion
# This is the container that Twoliter uses to invoke `cargo make` and other build
# commands. It will be available for both x86_64 and Arm hosts, but it is agnostic
# as to which architecture the build is targeting.
[build-image]
registry = "public.ecr.aws/bottlerocket"
name = "twoliter"
version = "v0.1.0"
```
```